src/store/GlobalContext.jsx

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+/* eslint-disable react/prop-types */
+import { createContext, useEffect, useState } from 'react';
+import { useCurrentLocation } from '../hooks/useCurrentLocation';
+import useWeather from '../hooks/useWeather';
+
+const GlobalContext = createContext();
+
+function GlobalProvider({ children }) {
+  const [weatherObj, setWeatherObj] = useState();
+  const [searchField, setSearchField] = useState(false);
+  const [selectedLocation, setSelectedLocation] = useState({
+    latitude: null,
+    longitude: null,
+    locationName: null,
+  });
+
+  const toggleSearchField = () => {
+    setSearchField(prevSearchField => !prevSearchField);
+  };
+
+  const handleGeolocate = async () => {
+    try {
+      const location = await useCurrentLocation();
+      setSelectedLocation(prevLocation => ({
```

Comment:
you don't need to call setter with callback here

it's useful when the next value is based on previous

Reply:
was initially planning to have more than just coordinates for location, and initially updating with previous state made sense. Not anymore.